tempo_embeddings/embeddings/vector_database.py

```diff
@@ -0,0 +1,336 @@
+from numpy.typing import ArrayLike
+
+from ..text.passage import Passage
+if TYPE_CHECKING:
+    from ..text.corpus import Corpus
+
+from transformers import AutoTokenizer
+
+import chromadb
+import chromadb.utils.embedding_functions as embedding_functions
+from chromadb.types import Collection
+from chromadb.db.base import UniqueConstraintError
+
+from umap.umap_ import UMAP
+
+
+class VectorDatabaseManagerWrapper(ABC):
+    """A Wrapper for different Vector Databases"""
+
+    def __init__(
+        self,
+        batch_size: int
+    ):
+        """Constructor.
+
+        Args:
+            batch_size: The batch size to process records
+        """
+        self._batch_size = batch_size
+
+    @property
+    def batch_size(self) -> int:
+        return self._batch_size
+
+    @batch_size.setter
+    def batch_size(self, batch_size: int) -> None:
+        self._batch_size = batch_size
+
+
+    def _batches(self, passages: list[Passage]) -> Iterable[list[Passage]]:
+        for batch_start in tqdm(
+            range(0, len(passages), self.batch_size),
+            desc="Embeddings Batches",
+            unit="batch",
+            total=len(passages) // self.batch_size + 1,
+        ):
+            yield passages[batch_start : batch_start + self.batch_size]
+
+    @abstractmethod
+    def connect(self):
+        pass
+
+    @abstractmethod
+    def insert_passages_embeddings(self, collection:Collection, passages: List[Passage]):
+        pass
+
+    @abstractmethod
+    def retrieve_vectors_if_exist(self, collection: Collection, passages: List[Passage]):
+        pass
+
+
+class ChromaDatabaseManager(VectorDatabaseManagerWrapper):
+    """A Chroma Database can create N collections and will always use THE SAME embedder function and tokenizer for all collections.
+        To create collections with different embedders one needs to create separate databases
+    """
+    def __init__(self, 
+                 db_path: str = "default_db",
+                 embedder_name: str = "sentence-transformers/all-MiniLM-L6-v2",
+                 embedder_config: Dict[str,Any] = None,
+                 batch_size: int = 8):
+        super().__init__(batch_size)
+        self.db_path = db_path
+        self.embedder_name = embedder_name
+        self.embedder_config = embedder_config
+        self.tokenizer = AutoTokenizer.from_pretrained(embedder_name)
+        self.client =  None
+        self.requires_explicit_embeddings = False
+        if self.embedder_config and self.embedder_config.get("type") == "hf":
+            self.embedding_function = embedding_functions.HuggingFaceEmbeddingFunction(
+                api_key=self.embedder_config.get("api_key"),
+                model_name=embedder_name
+            )
+        elif self.embedder_config and self.embedder_config.get("type") == "custom":
+            self.requires_explicit_embeddings = True
+            self.embedding_function = None
+        else:
+            self.embedding_function = None
```

Comment:
1. The `self.embedder_config.get("type")` call makes the `if self.embedder_config` condition redundant, because it returns `None` by default.
2. This logic seems to silently ignore cases where an invalid/unrecognized value for `self.embedder_config` has been given, which could result in unexpected behaviour in case of user typos etc.